### PR TITLE
Minor update to water heater installation location default logic to be consistent with ANSI 301 2022.

### PR DESCRIPTION
**Pull request by jmaguire1**
## Pull Request Description

Update our default WH installation location logic to match what's in the latest ANSI 301 standard. Really minor changes overall, we need to change how we treat CZ 3A (as a "hot" rather than "cold" location) and make unconditioned basements the priority if a home has both a conditioned and unconditioned basement.

There's a second part to this about the reference home, I'll do that in a separate PR off the "ansi_301_2022" branch. This change felt like one we'd want to make to our defaulting across the board. Our logic was originally based on BA climate zones rather than IECC and that's how we ended up slightly different, but now that someone else has weighed in on this it makes sense to default to a logic we can cite.

## Checklist

PR Author: Check these when they're done. Not all may apply. ~~strikethrough~~ and check any that do not apply. 

PR Reviewer: Verify each has been completed.

~~- [ ] Schematron validator (`EPvalidator.xml`) has been updated~~
~~- [ ] Sample files have been added/updated (`openstudio tasks.rb update_hpxmls`)~~
~~- [ ] Tests have been added/updated (e.g., `HPXMLtoOpenStudio/tests/test*.rb` and/or `workflow/tests/test*.rb`)~~
- [X] Documentation has been updated
- [X] Changelog has been updated
- [X] `openstudio tasks.rb update_measures` has been run
- [X] No unexpected changes to simulation results of sample files

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ __New Features__
   - Adds detailed performance data inputs for variable-speed air source HVAC systems.
   - Add soil and moisture type arguments (for determining ground conductivity and diffusivity) and optional geothermal loop arguments for ground source heat pumps.
   - The "Geometry: Building Number of Units" input is now written to the HPXML `NumberofUnitsInBuilding` element.
+  - Updated water heater installation location defaulting to match ANSI 301-2022
```
Comment (shorowit) [inline Changelog.md:17]: Need to pull this out one level; this change is not specific to the BuildResidentialHPXML measure.

```suggestion
- Updated water heater installation location defaulting to match ANSI 301-2022
```

File: docs/source/workflow_inputs.rst [hunk 2]
```diff
@@ -3260,9 +3260,9 @@ Each conventional storage water heater is entered as a ``/HPXML/Building/Buildin
          See :ref:`hpxmllocations` for descriptions.
   .. [#] If Location not provided, defaults to the first present space type:
          
-         \- **IECC zones 1-3, excluding 3A**: "garage", "conditioned space"
+         \- **IECC zones 1-3**: "garage", "conditioned space"
          
-         \- **IECC zones 3A, 4-8, unknown**: "basement - conditioned", "basement - unconditioned", "conditioned space"
+         \- **IECC zones 3-8, unknown**: "basement - unconditioned", "basement - conditioned", "conditioned space"
```
Comment (shorowit) [inline docs/source/workflow_inputs.rst:3265]: Good change. This logic is (for better or worse) repeated for each water heater type, can you make the same change to all of them?

File: workflow/tests/base_results/results_workflow_simulations2.csv [hunk 3]
```diff
@@ -115,7 +115,7 @@ house023.xml,138.233,138.233,63.089,63.089,0.0,75.144,0.0,0.0,0.0,0.0,0.0,1.892,
...
+house027.xml,73.458,73.458,31.285,31.285,42.173,0.0,0.0,0.0,0.0,0.0,0.0,0.463,0.0,0.0,7.475,0.941,0.0,0.0,0.0,5.947,0.218,0.485,0.927,0.0,0.0,0.0,1.724,0.0,0.0,0.447,0.338,2.514,0.105,0.0,2.114,7.587,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,21.172,0.0,17.932,0.0,3.069,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,19.943,0.0,21.86,8.564,5.271,0.0,0.0,0.0,0.0,1581.0,3606.3,3606.3,24.164,22.774,0.721,1.786,7.94,0.454,0.0,0.594,4.989,-4.006,0.0,0.0,0.321,3.265,-0.167,1.463,0.0,10.465,0.0,2.143,-7.25,-2.861,0.501,1.161,0.793,0.063,0.0,-0.105,0.214,5.405,0.0,0.0,0.149,3.861,-0.167,-0.276,-0.891,-3.327,0.0,2.467,8.837,3.086,1610.9,1574.7,10579.7,3728.4,0.0,75000.0,36000.0,0.0,24.62,91.58,33087.0,7577.0,4494.0,0.0,375.0,6506.0,550.0,250.0,4088.0,1516.0,7731.0,19084.0,3678.0,4014.0,0.0,228.0,2868.0,270.0,163.0,0.0,2277.0,2267.0,3320.0,5496.0,1758.0,2939.0,800.0
```
Comment (shorowit) [inline workflow/tests/base_results/results_workflow_simulations2.csv:118]: Confirmed that house027.xml is in climate zone 3A (so the water heater now ends up in the garage).

**Review by shorowit — COMMENTED** [review]
Looks good, @jmaguire1! Just two small changes are needed and then this can be merged in.

File: docs/source/workflow_inputs.rst (hunk 2 above)
Comment (jmaguire1) [reply docs/source/workflow_inputs.rst:3265]: Just fixed in the latest commit. Thanks for pointing it out!

**Review by shorowit — APPROVED** [review]

**Pull request merged**